src/cdk-experimental/popover-edit/edit-ref.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Subject} from 'rxjs';
+import {first} from 'rxjs/operators';
+import {Injectable, OnDestroy, Self} from '@angular/core';
+import {ControlContainer} from '@angular/forms';
+import {EditEvents} from './edit-events';
+
+/**
+ * Service used for communication between the form within the edit lens and the
+ * table that launched it. Provided by CdkPopoverEditControl within the lens.
+ */
+@Injectable()
+export class EditRef implements OnDestroy {
+  /** Emits the final value of this edit instance before closing. */
+  private readonly _finalValueSubject = new Subject<any>();
+  readonly finalValue = this._finalValueSubject.asObservable();
+
+  /** The value to set the form back to on revert. */
+  private _revertFormValue: any;
```

Comment:
Could this class use a generic for the value type throughout the class? We do something similar for `MatDialogRef`